File: jjx-web/src/enums/production/EquipmentStatusEnum.ts
import { createNamedEnum } from '../base'

/** 生产设备运行状态，对应后端 EquipmentStatusEnum。 */
export const EquipmentStatusEnum = createNamedEnum(
  {
    STANDBY: { value: 0, label: '待机中', tagProps: { type: 'info' } },
    RUNNING: { value: 1, label: '运行中', tagProps: { type: 'success' } },
    MAINTENANCE: { value: 2, label: '维护中', tagProps: { type: 'warning' } },
    FAULT: { value: 3, label: '故障中', tagProps: { type: 'danger' } },
  },
  { type: 'info' }
)

export const AVAILABLE_EQUIPMENT_STATUSES = [
  EquipmentStatusEnum.STANDBY.value,
  EquipmentStatusEnum.RUNNING.value,
] as const

File: jjx-server/src/main/java/com/jjx/production/enums/EquipmentStatusEnum.java
package com.jjx.production.enums;

import lombok.Getter;

/** 生产设备运行状态。 */
@Getter
public enum EquipmentStatusEnum {
    STANDBY(0, "待机中", true),
    RUNNING(1, "运行中", true),
    MAINTENANCE(2, "维护中", false),
    FAULT(3, "故障中", false);

    private final int value;
    private final String label;
    private final boolean available;

    EquipmentStatusEnum(int value, String label, boolean available) {
        this.value = value;
        this.label = label;
        this.available = available;
    }

    public static EquipmentStatusEnum getByValue(Integer value) {
        if (value == null) return null;
        for (EquipmentStatusEnum status : values()) {
            if (status.value == value) return status;
        }
        return null;
    }
}
